mmeval/metrics/cityscapes_detection.py

```diff
@@ -0,0 +1,327 @@
+                  scores of masks.
+
+            groundtruths (Sequence[dict]): A sequence of dict. If load from
+                `ann_file`, the dict inside can be empty. Else, each dict
+                represents a groundtruths for an image, with the following
+                keys:
+
+                - file_name (str): The absolute path of groundtruth image.
+        """
+        for prediction, groundtruth in zip(predictions, groundtruths):
+            assert isinstance(prediction, dict), 'The prediciton should be ' \
+                f'a sequence of dict, but got a sequence of {type(prediction)}.'  # noqa: E501
+            assert isinstance(groundtruth, dict), 'The label should be ' \
+                f'a sequence of dict, but got a sequence of {type(groundtruth)}.'   # noqa: E501
+            prediction = self._process_prediction(prediction)
+            self._results.append((prediction, groundtruth))
+
+    def compute_metric(self, results: list) -> Dict[str, Union[float, list]]:
+        """Compute the CityScapes metrics.
+
+        Args:
+            results (List[tuple]): A list of tuple. Each tuple is the
+                prediction and ground truth of an image. This list has already
+                been synced across all ranks.
+
+        Returns:
+            dict: The computed metric. The keys are the names of
+            the metrics, and the values are corresponding results.
+        """
+        eval_results: OrderedDict = OrderedDict()
+
+        if self.format_only:
+            self.logger.info(
+                f'Results are saved in {osp.dirname(self.outfile_prefix)}')
+            return eval_results
+        gt_instances_file = osp.join(self.outfile_prefix, 'gtInstances.json')
+        # split gt and prediction list
+        preds, gts = zip(*results)
+        CSEval.args.cityscapesPath = osp.join(self.seg_prefix, '..', '..')  # type: ignore # yapf: disable # noqa: E501
+        CSEval.args.predictionPath = self.outfile_prefix
+        CSEval.args.predictionWalk = None
+        CSEval.args.JSONOutput = False
+        CSEval.args.colorized = False
+        CSEval.args.gtInstancesFile = gt_instances_file
+
+        groundTruthImgList = [gt['file_name'] for gt in gts]
+        predictionImgList = [pred['pred_txt'] for pred in preds]
+        CSEval_results = CSEval.evaluateImgLists(predictionImgList,
+                                                 groundTruthImgList,
+                                                 CSEval.args)['averages']
+
+        # If gt_instances_file exists, CHEval will load it on the next
+        # CSEval.evaluateImgLists without recalculating.
+        if not self.keep_gt_json:
+            os.remove(gt_instances_file)
+
+        if not self.keep_results:
+            shutil.rmtree(self.outfile_prefix)
+
+        # remove `matches.json` file
+        file_path = osp.join(os.getcwd(), 'matches.json')
+        if osp.exists(file_path):
+            os.remove(file_path)
```

Comment:
These operations are very dangerous in distributed environment.
Should work out a solution here.